File: src/Java/myservicebus/src/main/java/com/myservicebus/DefaultBusFactory.java
package com.myservicebus;

import java.util.function.Consumer;

public class DefaultBusFactory implements BusFactory {
    @Override
    public <T extends BusFactoryConfigurator> MessageBus create(Class<T> configuratorClass, Consumer<T> configure) {
        try {
            T cfg = configuratorClass.getDeclaredConstructor().newInstance();
            if (configure != null) {
                configure.accept(cfg);
            }
            return cfg.build();
        } catch (Exception e) {
            throw new RuntimeException("Unable to create configurator", e);
        }
    }
}

File: src/MyServiceBus/DefaultBusFactory.cs
namespace MyServiceBus;

using System;

public class DefaultBusFactory : IBusFactory
{
    public IMessageBus Create<T>(Action<T>? configure = null)
        where T : IBusFactoryConfigurator, new()
    {
        var cfg = new T();
        configure?.Invoke(cfg);
        return cfg.Build();
    }
}
